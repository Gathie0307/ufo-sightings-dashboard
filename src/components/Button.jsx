const Button = ({ text, disabled, handler }) => {
  return (
    <button
      className='rounded-full w-[150px] px-4 py-2 bg-blue-500 text-white hover:bg-blue-600 cursor-pointer disabled:cursor-not-allowed disabled:bg-gray-300 disabled:text-gray-500'
      disabled={disabled}
      onClick={handler}
    >
      {text}
    </button>
  );
};

export default Button;
