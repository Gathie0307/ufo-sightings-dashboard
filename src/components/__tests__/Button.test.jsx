import { render } from '@testing-library/react';
import Button from '../Button';

describe('Button', () => {
  it('should match enabled snapshot', () => {
    const { asFragment } = render(<Button disabled={false} text='Previous' />);

    expect(asFragment()).toMatchSnapshot();
  });

  it('should match disabled snapshot', () => {
    const { asFragment } = render(<Button disabled={true} text='Next' />);

    expect(asFragment()).toMatchSnapshot();
  });
});
