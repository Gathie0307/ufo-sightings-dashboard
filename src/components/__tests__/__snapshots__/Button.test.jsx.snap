// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button > should match disabled snapshot 1`] = `
<DocumentFragment>
  <button
    class="rounded-full w-[150px] px-4 py-2 bg-blue-500 text-white hover:bg-blue-600 cursor-pointer disabled:cursor-not-allowed disabled:bg-gray-300 disabled:text-gray-500"
    disabled=""
  >
    Next
  </button>
</DocumentFragment>
`;

exports[`Button > should match enabled snapshot 1`] = `
<DocumentFragment>
  <button
    class="rounded-full w-[150px] px-4 py-2 bg-blue-500 text-white hover:bg-blue-600 cursor-pointer disabled:cursor-not-allowed disabled:bg-gray-300 disabled:text-gray-500"
  >
    Previous
  </button>
</DocumentFragment>
`;
